src/nbl/video/CVulkanQueue.cpp

```diff
@@ -103,15 +104,52 @@ bool CVulkanQueue::submit(uint32_t _count, const SSubmitInfo* _submits, IGPUFenc
     }
 
     VkFence fence = _fence ? static_cast<CVulkanFence*>(_fence)->getInternalObject() : VK_NULL_HANDLE;
-    vk->vk.vkQueueSubmit(m_vkqueue, _count, submits, fence);
+    // vk->vk.vkQueueSubmit(m_vkqueue, _count, submits, fence);
+    if (vkQueueSubmit(m_vkQueue, _count, submits, fence) == VK_SUCCESS)
+        return true;
 
-    // Todo(achal): THIS IS FAKE !!!!
     return false;
 }
 
 bool CVulkanQueue::present(const SPresentInfo& info)
 {
-    return false;
+    assert(info.waitSemaphoreCount <= 100);
+    VkSemaphore vk_waitSemaphores[100];
+    for (uint32_t i = 0u; i < info.waitSemaphoreCount; ++i)
+    {
+        if (info.waitSemaphores[i]->getAPIType() != EAT_VULKAN)
+            return false;
+
+        vk_waitSemaphores[i] = reinterpret_cast<CVulkanSemaphore*>(info.waitSemaphores[i])->getInternalObject();
+    }
+
+    assert(info.swapchainCount <= 5);
+    VkSwapchainKHR vk_swapchains[5];
+    for (uint32_t i = 0u; i < info.swapchainCount; ++i)
+    {
+        if (info.swapchains[i]->getAPIType() != EAT_VULKAN)
+            return false;
+
+        vk_swapchains[i] = reinterpret_cast<CVKSwapchain*>(info.swapchains[i])->m_vkSwapchainKHR;
+    }
+
+    VkPresentInfoKHR presentInfo = { VK_STRUCTURE_TYPE_PRESENT_INFO_KHR };
+    presentInfo.waitSemaphoreCount = info.waitSemaphoreCount;
+    presentInfo.pWaitSemaphores = vk_waitSemaphores;
+    presentInfo.swapchainCount = info.swapchainCount;
+    presentInfo.pSwapchains = vk_swapchains;
+    presentInfo.pImageIndices = info.imgIndices;
+
+    VkResult result = vkQueuePresentKHR(m_vkQueue, &presentInfo);
+
+    switch (result)
+    {
+    case VK_SUCCESS:
+    case VK_SUBOPTIMAL_KHR:
+        return true;
+    default:
+        return false;
+    }
```

Comment:
make an enum in `ISwapchain` for this and return the full enum

Reply:
Resolved